datatypes/src/lib.rs

```diff
@@ -34,3 +34,19 @@ pub mod util;
 /// Compiles Geo Engine Pro
 #[cfg(feature = "pro")]
 pub mod pro;
+
+#[macro_export]
+macro_rules! test_data {
```

Comment:
as i understand it: the macro is executed at compile time. that means the code ins inserted and evaluated where the macro is located. That will cause the env variable in the macro to correspond to the crates path. It might happen that you only have one crate where the macro is used but not defined. Then the call to canonicalize().parent() will operate on a folder that does not exist and i guess this can fail. Also it is like this in the rust-gdal crate and that was my example.